pkg/artifacts/registry.go

```diff
@@ -2,42 +2,57 @@ package artifacts
 
 import (
 	"context"
-	"fmt"
 
-	"oras.land/oras-go/pkg/content"
-	"oras.land/oras-go/pkg/oras"
+	"github.com/go-logr/logr"
+
+	"github.com/aws/eks-anywhere-packages/pkg/registry"
 )
 
+const certFile = "/tmp/registry-mirror/CACERTCONTENT"
+
 // RegistryPuller handles pulling OCI artifacts from an OCI registry
 // (i.e. bundles)
-type RegistryPuller struct{}
+type RegistryPuller struct {
+	storageClient registry.StorageClient
+	log           logr.Logger
+}
 
 var _ Puller = (*RegistryPuller)(nil)
 
 // NewRegistryPuller creates and initializes a RegistryPuller.
 //
 // It assumes AWS ECR, and uses a password that exists in the ECR_PASSWORD
 // environment variable.
-func NewRegistryPuller() *RegistryPuller {
-	return &RegistryPuller{}
+func NewRegistryPuller(logger logr.Logger) *RegistryPuller {
+	return &RegistryPuller{
+		log: logger,
+	}
 }
 
 func (p *RegistryPuller) Pull(ctx context.Context, ref string) ([]byte, error) {
-	registry, err := content.NewRegistry(content.RegistryOptions{})
+	var art registry.Artifact
+	err := art.SetURI(ref)
 	if err != nil {
-		return nil, fmt.Errorf("creating registry: %w", err)
+		return nil, err
 	}
-	store := content.NewMemory()
 
-	_, err = oras.Copy(ctx, registry, ref, store, "")
+	certificates, err := registry.GetCertificates(certFile)
 	if err != nil {
-		return nil, fmt.Errorf("pulling artifact %q: %s", ref, err)
+		p.log.Error(err, "problem getting certificate file", "filename", certFile)
```

Comment:
shall we return from this method right after logging?

Reply:
No, best effort if there is no file. We don't need certs for ECR for example, so that will still work.